src/VisualStudio/Core/Test.Next/Services/AssetStorageTests.cs

```diff
@@ -47,17 +43,30 @@ public void TestGetAssets()
             Assert.True(storage.TryGetAsset(checksum, out stored));
         }
 
-        private class MyAssetSource : AssetSource
+        [Fact, Trait(Traits.Feature, Traits.Features.RemoteHost)]
+        public async Task TestCleanup()
         {
-            public MyAssetSource(AssetStorage assetStorage, int sessionId) :
-                base(assetStorage, sessionId)
-            {
-            }
+            var storage = new AssetStorage(cleanupInterval: TimeSpan.FromMilliseconds(1), purgeAfter: TimeSpan.FromMilliseconds(2));
+
+            var checksum = new Checksum(Guid.NewGuid().ToByteArray());
+            var data = new object();
+
+            Assert.True(storage.TryAddAsset(checksum, data));
 
-            public override Task<IList<(Checksum, object)>> RequestAssetsAsync(int serviceId, ISet<Checksum> checksums, CancellationToken cancellationToken)
+            for (var i = 0; i < 10; i++)
             {
-                throw new NotImplementedException();
+                await Task.Delay(10);
```

Comment:
actually our IAsyncToken thing is not right one to use here. this is not tracking async operations.

sure it can take up to 100ms to find out test is failed. and that is actually better than waiting for event in failed case since that will cause whole jenkin to time out.

usually worry is on false positive (failing when it is actually working correctly) but since it depends on same task for timing (cache cleanup and wait on test), I dont think false positive will happen. and we can address that if it ever actually happen.